pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -77,6 +76,32 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if subnetPort.ObjectMeta.DeletionTimestamp.IsZero() {
+		if len(subnetPort.Spec.Subnet) != 0 {
+			s := &v1alpha1.Subnet{}
+			if err := r.Client.Get(ctx, req.NamespacedName, s); err != nil {
+				log.Error(err, "unable to fetch subnet CR", "req", req.NamespacedName)
+				return common.ResultRequeueAfter10sec, err
+			}
+			if s != nil && !s.DeletionTimestamp.IsZero() {
+				err := fmt.Errorf("subnet %s is been deleting, cannot operate subnetport %s", s.Name, req.NamespacedName)
+				updateFail(r, ctx, subnetPort, &err)
+				return common.ResultNormal, err
+			}
+		}
+
+		if len(subnetPort.Spec.SubnetSet) != 0 {
```

Comment:
if there is no Subnet or SubnetSet, it will use default SubnetSet for both VM and Pod.

Reply:
updated